### PR TITLE
 Timestamps added and min/max removed from templates

### PR DESCRIPTION
**Pull request by djesic**

### CODE REVIEW

**Review by nvuletic — APPROVED** [review]

**Review by LazarBozic94 — APPROVED** [review]

File: test/test_wolk.py [hunk 1]
```diff
@@ -440,6 +417,281 @@ def test_add_simple_device(self):
...
+                firmware_update_protocol=False,
+            )
+
+    def test_good_firmware_update_protocol(self):
+        """Test passing a good firmware update protocol doesn't raise an exception."""
+        firmware_update_protocol = JsonFirmwareUpdateProtocol()
+
+        wolk = Wolk(
+            "host",
+            1883,
```
Comment (srdjastankovic) [inline test/test_wolk.py:585]: why test connection with unsecure port?

File: wolk_gateway_module/connectivity/connectivity_service.py [hunk 2]
```diff
@@ -74,22 +74,22 @@ def connected(self) -> bool:
...
     def reconnect(self) -> Optional[bool]:
         """Reestablish connection with WolkGateway."""
-        pass
+        raise NotImplementedError
 
     @abstractmethod
     def disconnect(self) -> Optional[bool]:
         """Terminate connection with WolkGateway."""
-        pass
+        raise NotImplementedError
```
Comment (srdjastankovic) [inline wolk_gateway_module/connectivity/connectivity_service.py:92]: how to implement error? eg NotImplementedError = "some error" in main module?

File: test/test_wolk.py (hunk 1 above)
Comment (djesic) [reply test/test_wolk.py:585]: No connection is established, this is just to satisfy the constructor

File: wolk_gateway_module/connectivity/connectivity_service.py (hunk 2 above)
Comment (djesic) [reply wolk_gateway_module/connectivity/connectivity_service.py:92]: No, that is the name of the error - NotImplemented

Comment (djesic) [reply wolk_gateway_module/connectivity/connectivity_service.py:92]: builtin exception

Comment (srdjastankovic) [reply wolk_gateway_module/connectivity/connectivity_service.py:92]: Is it necessary to raise exceptions? This is controlled exit

Comment (djesic) [reply wolk_gateway_module/connectivity/connectivity_service.py:92]: It's an abstract method, so yes

Comment (srdjastankovic) [reply wolk_gateway_module/connectivity/connectivity_service.py:92]: so what

Comment (djesic) [reply wolk_gateway_module/connectivity/connectivity_service.py:92]: Attempting to call an abstract method should not just "pass", but raise an exception. Especially if the method is expected to return a value.

Comment (srdjastankovic) [reply wolk_gateway_module/connectivity/connectivity_service.py:92]: You answered it, return a value.

Comment (djesic) [reply wolk_gateway_module/connectivity/connectivity_service.py:92]: Regardless of what is in the body of this method, either `pass` or `raise NotImplementedError` - if this method is not implemented for this class, an exception will be raised as `@abstractmethod` decorator required it to be implemented for this class.

This whole change is here to be more explicit, the body of it won't actually be run at any time.

**Review by srdjastankovic — APPROVED** [review]

**Comment by djesic** [comment]
Leaving this open for a bit longer - need to apply reconnection fix for MQTT like the one in WC-Py

**Pull request merged**